android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonFormFragment.java

```diff
@@ -271,6 +271,146 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    /**
+     * Skips blank by relevance steps when next is clicked on the json wizard forms.
+     */
+    public void skipLoadedStepsOnNextPressed() {
+        if (skipBlankSteps()) {
+            JSONObject formStep = getStep(getArguments().getString(JsonFormConstants.STEPNAME));
+            String next = formStep.optString(JsonFormConstants.NEXT, "");
+            if (StringUtils.isNotEmpty(next)) {
+                checkIfStepIsBlank(formStep);
+                if (shouldSkipStep() && !stepHasNoSkipLogic(JsonFormConstants.STEP1)) {
+                    getJsonApi().setNextStep(next);
+                    markStepAsSkipped(formStep);
+                    next();
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the current form step number when given than steps next step number.
+     * This number is used to figure out which steps to pop when previous is clicked.
+     *
+     * @param nextFormNumber {@link String}
+     * @return formNumber {@link Integer}
+     */
+    private int getFormStepNumber(String nextFormNumber) {
+        int formNumber = 0;
+        if (StringUtils.isNotBlank(nextFormNumber)) {
+            int currentFormNumber = Integer.parseInt(nextFormNumber.substring(4, 5)) - 1;
```

Comment:
I think `nextFormNumber.substring(4, 5)` should be   `nextFormNumber.substring(4)` instead. Take a form with two digit steps e.g. `step10` for example.

Reply:
Also, do we need this complex logic to extract the current step? Wasn't this working? https://github.com/OpenSRP/opensrp-client-native-form/pull/407/files#diff-4af3b3a5f5179b96a2f8051da8ad980ec341e72fc18aa4bd23f0cbc2c2216845L738?

Reply:
I don't think there's a constraint on NF that guarantees a step's `next` value has to be +1 of it's current value. In RDT, we have steps that don't even have a next value.